Hypothesis Testing assignment with Customer order Form.csv.
H0 = Customer order forms defective % does not varies by centre
H1 = customer order forms defective % does vary by centre

We are doing an Chi square the data is catagorical.
And comparing the p value with alpha.

In [2]:
import pandas as pd
import numpy as np
import scipy
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency
data=pd.read_csv('Costomer+OrderForm.csv')
data.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [3]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [4]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [5]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [6]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [7]:
# Make a contingency table
obs=np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [8]:
# Chi2 contengency independence test
chi2_contingency(obs) # o/p is (Chi2 stats value, p_value, df, expected obsvations)

Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.2771020991233135, dof=3, expected_freq=array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))

Inference: As (p_value = 0.2771) > (α = 0.05); Accept Null Hypthesis i.e. Independence of categorical variables Thus, customer order forms defective % does not vary by centre.